src/Database/Expression/AggregateExpression.php

```diff
@@ -87,7 +87,9 @@ public function over(?string $name = null)
     public function partition($partitions)
     {
         $this->over();
-        $this->window->partition($partitions);
+        /** @var \Cake\Database\Expression\WindowExpression $window */
+        $window = $this->window;
+        $window->partition($partitions);
```

Comment:
This is an annoying workaround for psalm since it can't handle a function initializing a property.